### PR TITLE
Statistics finder email signup page

### PR DESCRIPTION
**Pull request by bilbof**
https://trello.com/c/mMPcqr2x/509-default-finder-radio-button-should-always-be-applied
https://trello.com/c/5oag3R25/288-publish-statistics-finder-from-rummager-s

This is the final step to enable us to publish the new stats finder.

Changes:
- Adds email signup page for testing
- Bugfix: the Radio facet & filter will now use a provided default value when unset
- Bugfix: the query strings for email alerts and atom feeds will be correct on first page load.

Reviewing by commit will be easier - see commit messages.

### CODE REVIEW

File: spec/lib/filters/radio_filter_spec.rb [hunk 1]
```diff
@@ -43,39 +94,73 @@
...
+
+      context "when no option is selected" do
+        it "should return the corresponding default values from the option_lookup" do
+          expect(radio_filter.value).to eq(%w(guidance))
+        end
+      end
+
+      context "when a disallowed value is provided" do
+        let(:params) { "does_not_exist" }
+        it "should return the corresponding default values from the option_lookup" do
```
Comment (karlbaker02) [inline spec/lib/filters/radio_filter_spec.rb]: Is it worth again here splitting out into a default value being provided and not being provided as you have above?

**Comment by karlbaker02** [comment]
Is it also worth adding an explicit test for a radio with previously selected value that is then reset to `nil` if for example it is set to `nil` via the facet (is this actually a behaviour that the radio facet should/will support)?

File: app/models/radio_facet.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class RadioFacet < FilterableFacet
+  def value
+    @value || default_value['value']
```
Comment (karlbaker02) [inline app/models/radio_facet.rb]: Where is `@value` defined? There's a `value` on the `FilterableFacet` base class, but I don't think it's this?

File: app/lib/filters/radio_filter.rb [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+
+      Array(default_allowed_value)
+    end
+
+    def option_lookup_values(val)
+      option_lookup.select { |key, _| val.include? key }.values.flatten
+    end
+
+    def default_allowed_value
+      @default_allowed_value ||= facet['allowed_values'].find(Proc.new { {} }) { |option| option['default'] }
```
[the remark this answers is not in the document]
Comment (bilbof) [reply app/lib/filters/radio_filter.rb]: It sets a default return value of `{}` rather than the default `nil`. This is instead of doing `|| {}` afterwards. https://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-find

File: app/lib/filters/radio_filter.rb [hunk 4]
```diff
@@ -1,7 +1,36 @@
...
+      @default_allowed_value['value']
+    end
+
+    def acceptable_param?
+      params.present? && params.is_a?(String) && param_is_part_of_allowed_values
+    end
+
+    def param_is_part_of_allowed_values
+      facet['allowed_values'].find { |option| option['value'] == params }.present?
```
[the remark this answers is not in the document]
Comment (bilbof) [reply app/lib/filters/radio_filter.rb]: Yeah we can, thanks!

File: app/models/radio_facet.rb (hunk 2 above)
Comment (bilbof) [reply app/models/radio_facet.rb]: It's quite hard to see. It's set within `FacetCollection` here:

https://github.com/alphagov/finder-frontend/blob/master/app/models/facet_collection.rb#L15

This gets called in the `FinderPresenter`:

https://github.com/alphagov/finder-frontend/blob/master/app/presenters/finder_presenter.rb#L16

This is how values from the params are translated into their corresponding facets.

**Comment by bilbof** [comment]
Currently there is no case where a user can reset the radio button to nil. I don't think that the facet needs to support this yet, though it may support it now.

File: spec/lib/filters/radio_filter_spec.rb (hunk 1 above)
Comment (bilbof) [reply spec/lib/filters/radio_filter_spec.rb]: Yes good idea. And actually, this unearthed a bug - see the last commit. It was unexpected behaviour for there to be no default value and an option_lookup. Thanks for suggesting this.
